alpheios_nemo_ui/__init__.py

```diff
@@ -426,6 +427,19 @@ def r_userinfo(self):
     def r_usertoken(self):
         return jsonify(session['access_token'])
 
+    def r_next_passage(self, objectId, subreference):
```

Comment:
I am guessing you created a separate route for this to have a nice JSON object with the data -- technically it's all available in the page already, and could be pulled together by the template code rather than requiring a separate ajax call for it.  Either is probably ok, this is cleaner code certainly, maybe just a little less efficient because of the extra network call.

Reply:
When I start to check the `mobile.js` and existed infinite scroll (I was able to reproduce conditions that it was working) - I saw that it loads the same and the same passage.
I checked the code and found that it took the next reference from the next link.
But it worked only the first time, when you uploaded the next passage, updates the subrefernce in the breadcrumb, next ref was not recalced (because it has python url method).

That's why the page has data only for one next step.
And that's why I created the most light way to get additional data (the next after next with python url method and so on) - I believe that :)

If I missed the place where all next data (all next refferences for the infinite scroll) is placed, I would be glad to get your help to see it!

Reply:
well, I was doing it a kind of hacky way, using the `updateNext` and `updatePrev` methods in mobile.js to update the next and previous links each time a passage was requested, replacing the passage in the url that was already there. Your approach  is definitely cleaner, so we can stick with that for now and change it later if there are performance issues.

Reply:
Oh, I see - you didn't really construct a url (using route manager of a flask - you simply overwrite the part of the url. 
But anyway - how do you get the the second time the next passage number? It is not saved to the properties.

Here it is (master branch):
You have this data for the start
![image](https://user-images.githubusercontent.com/12377640/57015831-398e6300-6c5a-11e9-80aa-1e0f8c9ced1e.png)

![image](https://user-images.githubusercontent.com/12377640/57015841-414e0780-6c5a-11e9-8686-afc5b2b147ca.png)

Then scroll
it is updated
![image](https://user-images.githubusercontent.com/12377640/57015872-6773a780-6c5a-11e9-98c6-cbccd87de726.png)

And then it stops updated - as the next is unknown (I think so)
![image](https://user-images.githubusercontent.com/12377640/57015888-7d816800-6c5a-11e9-9934-58dca7df5f39.png)

And it simply add the same next passage.

How could be get the next after the next without asking flask?

Reply:
Oh I see - you are parsing the data from preloadNext

Reply:
well, that's probably the cause of #106 :-)